compose/material/material/src/desktopMain/kotlin/androidx/compose/material/DesktopAlertDialog.desktop.kt

```diff
@@ -1,437 +0,0 @@
-            content = { content() }
-        )
-    }
-
-    @Composable
-    override fun AlertDialog(
-        onDismissRequest: () -> Unit,
-        shape: Shape,
-        modifier: Modifier,
-        content: @Composable (Modifier) -> Unit
-    ) {
-        // Popups on the desktop are by default embedded in the component in which
-        // they are defined and aligned within its bounds. But an [AlertDialog] needs
-        // to be aligned within the window, not the parent component, so we cannot use
-        // [alignment] property of [Popup] and have to use [Box] that fills all the
-        // available space. Also [Box] provides a dismiss request feature when clicked
-        // outside of the [AlertDialog] content.
-        Popup(
-            popupPositionProvider = object : PopupPositionProvider {
-                override fun calculatePosition(
-                    anchorBounds: IntRect,
-                    windowSize: IntSize,
-                    layoutDirection: LayoutDirection,
-                    popupContentSize: IntSize
-                ): IntOffset = IntOffset.Zero
-            },
-            focusable = true,
-            onDismissRequest = onDismissRequest,
-            onKeyEvent = {
-                if (it.type == KeyEventType.KeyDown && it.awtEventOrNull?.keyCode == KeyEvent.VK_ESCAPE) {
-                    onDismissRequest()
-                    true
-                } else {
-                    false
-                }
-            },
-        ) {
-            val scrimColor = Color.Black.copy(alpha = 0.32f) //todo configure scrim color in function arguments
-            Box(
-                modifier = Modifier
-                    .fillMaxSize()
-                    .background(scrimColor)
-                    .pointerInput(onDismissRequest) {
-                        detectTapGestures(onPress = { onDismissRequest() })
-                    },
-                contentAlignment = Alignment.Center
-            ) {
-                content(
-                    modifier
-                        .shadow(elevation = 24.dp, shape = shape)
-                        .pointerInput(onDismissRequest) {
-                            detectTapGestures(onPress = {
-                                // Workaround to disable clicks on Surface background
-                                // https://github.com/JetBrains/compose-jb/issues/2581
-                            })
-                        }
-                )
-            }
-        }
-    }
-}
-
-/**
- * Shows Alert dialog as undecorated draggable window.
- */
-@ExperimentalMaterialApi
-object UndecoratedWindowAlertDialogProvider : AlertDialogProvider {
-    @Deprecated("Will be removed in 1.5; use the overload that takes the dialog shape")
-    @OptIn(ExperimentalComposeUiApi::class)
-    @Composable
-    override fun AlertDialog(
-        onDismissRequest: () -> Unit,
-        content: @Composable () -> Unit
-    ) {
-        DialogWindow(
```

Comment:
The current plan is to document how to use `DialogWindow` without `AlertDialog`

cc @igordmn

Reply:
We already have docs describing how to use `DialogWindow` without alert dialog content.

Not sure, if showing material alert dialog in a separate window will be in demand, as it probably won't look good (but I am not sure). And if we provide such doc, we probably should expose something like `AlertDialogContent`, otherwise, users should copy-paste it.

Reply:
What about make something like `AlertDialogWindow`?

Reply:
I'm leaning toward removing it, as it doesn't comply any spec, doesn't look good without a shadow, so it won't be used by anyone.

<img width="585" alt="image" src="https://github.com/JetBrains/compose-multiplatform-core/assets/5963351/e61dfc98-0e32-4c5a-9e1e-99ee0ba33e86">

Reply:
Reverted